File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/ETertiaryLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object ETertiaryLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl33
            AppTheme.STANDARD -> R.color.cstl33
            AppTheme.BLUE -> R.color.cbl33
            AppTheme.GREEN -> R.color.cgl33
            AppTheme.RED -> R.color.crl33
            AppTheme.NOTHING -> R.color.cnol33
        }
    }
}

File: lib/layer2_core/light/e_tertiary_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class ETertiaryLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl33;
      case 'standard': return cstl33;
      case 'blue': return cbl33;
      case 'green': return cgl33;
      case 'red': return crl33;
      case 'nothing': return cnol33;
      default: return cstl33;
    }
  }
}
